### PR TITLE
docs: prompts / prompt hub PRD

### PR DESCRIPTION
**Pull request by mikeldking**
This is the scoping of the prompt hub (management) piece of Arize Phoenix.

### CODE REVIEW

File: internal_docs/specs/prompts.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+In the abstract a prompt template has:
+
+- A human-readable name (identifier)
+- A description / README (markdown)
+- A prompt type (Chat, String, ...)
+- A format (f-string / mustache)
+- A set of revisions to the prompt
+
+In addition a prompt template might have associated model configurations (e.g. the prompt only works under certain parameters). In this case the prompt might have a model configuration associated with it. This includes:
```
Comment (axiomofjoy) [inline internal_docs/specs/prompts.md:30]: Not sure I follow this part. When is this association needed?

Comment (mikeldking) [reply internal_docs/specs/prompts.md:30]: Which model the prompt template works with - e.g. is the prompt template designed to work with gpt-4o or gemini.

Comment (RogerHYang) [reply internal_docs/specs/prompts.md:30]: Another example is the value of `role`, which in the case of a tool result is `tool` for openai, but `user` for anthropic.

Comment (mikeldking) [reply internal_docs/specs/prompts.md:30]: > Another example is the value of `role`, which in the case of a tool result is `tool` for openai, but `user` for anthropic.

That is a very interesting edge-case. I'd lean towards the more specific (e.g. openai) but it's a tough wrinkle to tackle. Tool uniformity is scary for sure.

**Comment by mikeldking** [comment]
Whoops this landed on main by mistake. My bad. Gotta find a way to do this better.

**Pull request closed without merging**